models/comments.model.js

```diff
@@ -12,28 +12,30 @@ exports.insertComment = async (comment) => {
     });
   }
 
-  if (await checkUserExists(username)) {
-    return Promise.reject({
-      status: 404,
-      msg: "User not found",
-    });
-  }
+  // if (await checkUserExists(username)) {
+  //   return Promise.reject({
+  //     status: 404,
+  //     msg: "User not found",
+  //   });
+  // }
 
-  if (await checkArticleExists(article_id)) {
-    return Promise.reject({
-      status: 404,
-      msg: "Articles not found",
+  // if (await checkArticleExists(article_id)) {
+  //   return Promise.reject({
+  //     status: 404,
+  //     msg: "Articles not found",
+  //   });
+
+  return db
+    .query(
+      `INSERT INTO comments (body, author, article_id) VALUES ($1, $2, $3) RETURNING *`,
+      [body, username, article_id]
+    )
+    .then((result) => {
+      return result.rows[0];
+    })
+    .catch((err) => {
+      return { code: err.code, detail: err.detail };
```

Comment:
You should only have one catch, which is in your controller. That should then pass the error to the error handlers. 
Please refactor this to have a single catch at the end of the chain - the chain should end in the controller